### PR TITLE
plan, executor: support ordered index join.

### PR DESCRIPTION
**Pull request by hanfei1991**
1. let index join support `KeepOrder`
2. adjust profile by `expectedCnt`

@winoros @lamxTyler PTAL

### CODE REVIEW

File: plan/stats.go [hunk 1]
```diff
@@ -27,19 +27,27 @@ type statsProfile struct {
...
 	return profile
 }
 
 func (p *basePhysicalPlan) statsProfile() *statsProfile {
-	return p.basePlan.profile
+	profile := p.basePlan.profile
+	if p.expectedCnt > 0 && p.expectedCnt < profile.count {
+		factor := p.expectedCnt / profile.count
+		profile.count = p.expectedCnt
+		for i := range profile.cardinality {
```
Comment (lamxTyler) [inline plan/stats.go]: Why not use the function `collapse`?

File: plan/stats.go [hunk 2]
```diff
@@ -27,19 +27,27 @@ type statsProfile struct {
 }
 
 // collapse receives a selectivity and multiple it with count and cardinality.
```
Comment (lamxTyler) [inline plan/stats.go]: Update the comment.

File: plan/stats.go (hunk 1 above)
Comment (hanfei1991) [inline plan/stats.go]: a * b / a is not always equal to b for float number

File: plan/stats.go (hunk 2 above)
Comment (hanfei1991) [inline plan/stats.go]: I didn't change this function

**Comment by hanfei1991** [comment]
@winoros @lamxTyler PTAL

**Comment by winoros** [comment]
Jenkins CI failed @hanfei1991

**Pull request merged**